### ISSUE TITLE
.map() used in user defined function does not read in all the variables 

### ISSUE THREAD

**Issue by LloydLS** [josdejong/mathjs#1550]
Hi,

When using .map() in a user defined function, only the array gets passed through into .map(). 

Using a simplified example (my actual use-case is more complicated, and not necessary in order to demonstrate the issue):

```
myFunc(arr, val) = arr.map(f(x,i,a) = x * val)
myFunc([1,2,3], 10) // returns: "Error: Undefined symbol val" in v6.0.2
                                // but returns the expected result: [10, 20, 30] in v3.11.4 
```

However, if you define the other variables before the function, then it works as expected:

```
val = 10;
myFunc(arr, val) = arr.map(f(x,i,a) = x * val)
myFunc([1,2,3], 10) // returns the expected result: [10, 20, 30] in v6.0.2
```
Interestingly, it is ignoring the value of `val` from `myFunc()`, and just using the previously defined value, e.g.:

```
val = 10;
myFunc(arr, val) = arr.map(f(x,i,a) = x * val)
myFunc([1,2,3], 20) // returns: [10, 20, 30]; expected result: [20, 40, 60]
```

Potentially related to #1024 and #1088 ?

Thanks!

Chris

**Comment by josdejong** [comment]
Thanks for reporting Chris, I can indeed reproduce this issue. I expect it to be similar to #1088, something with passing the scopes of `myFunc` and the inner function in `map` correctly.

**Comment by josdejong** [comment]
Fixed via https://github.com/josdejong/mathjs/commit/deb7c05b53eac09cc678ab07cf156fa44e3a87eb (a one-liner), not yet published

**Comment by josdejong** [comment]
Fixed now in `v6.0.3`
